app/client/cli/servicer.go

```diff
@@ -0,0 +1,243 @@
+	}
+
+	if resp.JSON200 == nil {
+		return nil, fmt.Errorf("Error getting current session: Unexpected response %v", resp)
+	}
+
+	return resp.JSON200, nil
+}
+
+// REFACTOR: reuse this function in all the query commands
+func getCurrentHeight(ctx context.Context) (int64, error) {
+	client, err := rpc.NewClientWithResponses(remoteCLIURL)
+	if err != nil {
+		return 0, fmt.Errorf("Error getting current height: %w", err)
+	}
+	resp, err := client.GetV1QueryHeightWithResponse(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("Error getting current height: %w", err)
+	}
+	if resp.HTTPResponse.StatusCode != http.StatusOK {
+		return 0, fmt.Errorf("Error getting current height: Unexpected status code %d", resp.HTTPResponse.StatusCode)
+	}
+
+	if resp.JSON200 == nil {
+		return 0, fmt.Errorf("Error getting current height: Unexpected response %v", resp)
+	}
+
+	return resp.JSON200.Height, nil
+}
+
+// IMPROVE: [K8s][LocalNet] Publish Servicer(s) Host and Port as env. vars in K8s: similar to Validators: #823
+// REFACTOR: move package-level variables (e.g. remoteCLIURL) to a cli object and consider storing it in the context
+func sendTrustlessRelay(ctx context.Context, servicerUrl string, relay *rpc.RelayRequest) (*rpc.PostV1ClientRelayResponse, error) {
+	client, err := rpc.NewClientWithResponses(servicerUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	return client.PostV1ClientRelayWithResponse(ctx, *relay)
+}
+
+func buildRelay(payload string, appPrivateKey crypto.PrivateKey, session *rpc.Session, servicer *rpc.ProtocolActor) (*rpc.RelayRequest, error) {
+	// TECHDEBT: This is mostly COPIED from pocket-go: we should refactor pocket-go code and import this functionality from there instead.
+	relayPayload := rpc.Payload{
+		// INCOMPLETE(#803): need to unmarshal into JSONRPC and other supported relay formats once proto-generated custom types are added.
+		Jsonrpc: "2.0",
+		Method:  payload,
+		// INCOMPLETE: set Headers for HTTP relays
+	}
+
+	relayMeta := rpc.RelayRequestMeta{
+		BlockHeight: session.SessionHeight,
+		// TODO: Make Chain Identifier type consistent in Session and Meta use Identifiable for Chain in Session (or string for Chain in Relay Meta)
+		Chain: rpc.Identifiable{
+			Id: session.Chain,
+		},
+		ServicerPubKey: servicer.PublicKey,
+		// TODO(#697): Geozone
+	}
+
+	relay := &rpc.RelayRequest{
+		Payload: relayPayload,
+		Meta:    relayMeta,
+	}
+	// TECHDEBT: Evaluate which fields we should and shouldn't marshall when signing the payload
+	reqBytes, err := json.Marshal(relay)
+	if err != nil {
+		return nil, fmt.Errorf("Error marshalling relay request %v: %w", relay, err)
+	}
+	hashedReq := crypto.SHA3Hash(reqBytes)
+	signature, err := appPrivateKey.Sign(hashedReq)
+	if err != nil {
+		return relay, fmt.Errorf("Error signing relay: %w", err)
+	}
+	relay.Meta.Signature = hex.EncodeToString(signature)
+
+	return relay, nil
```

Comment:
I'm pretty sure we're going to hit problems/issues with this function, but we'll deal with it when we hit it in the E2E test

Reply:
Noted. Work on E2E tests will start shortly.